ipnsfs/dir.go

```diff
@@ -0,0 +1,340 @@
+package ipnsfs
```

Comment:
i cant check for correctness until i understand more all the things this package is supposed to handle. maybe write a package doc?